pkg/metricsservice/utils/tls.go

```diff
@@ -43,16 +54,89 @@ func LoadGrpcTLSCredentials(certDir string, server bool) (credentials.TransportC
 		return nil, fmt.Errorf("failed to add client CA's certificate")
 	}
 
-	// Load certificate and private key
-	cert, err := tls.LoadX509KeyPair(path.Join(certDir, "tls.crt"), path.Join(certDir, "tls.key"))
+	// Load initial certificate and private key
+	mTLSCertificate, err := tls.LoadX509KeyPair(certPath, keyPath)
 	if err != nil {
 		return nil, err
 	}
 
+	// Start the watcher for cert updates
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+	err = watcher.Add(certDir)
+	if err != nil {
+		return nil, err
+	}
+
+	certMutex := sync.RWMutex{}
+	go func() {
+		log.V(1).Info("starting mTLS certificates monitoring")
+		for {
+			select {
+			case event, ok := <-watcher.Events:
+				if !ok { // Channel was closed (i.e. Watcher.Close() was called).
+					log.Error(err, "watcher stopped")
+					return
+				}
+				// We are only interested on rename changes on ..data dir
+				// as kubernetes creates first a temp folder with the new
+				// cert and then rename the whole folder
+				if !event.Has(fsnotify.Rename) ||
+					strings.HasSuffix(event.Name, "..data") {
+					continue
+				}
```

Comment:
not sure I'm reading this correctly, this will exclude `*..data` dir changes while the comment says we are interested in `..data` rename changes

Reply:
you're 100% right, good catch!. I's worked because rename events triggers also the change on the old folder, but using the right one is the way to go 😄 
I've updated my PR and I'm testing it right now, if tomorrow everything works and you agree, I'll merge the PR

Reply:
perfect, all good from my perspective, feel free to merge at your convenience